### PR TITLE
Use FieldDescriptorProto.json_name for representing field names.

### PR DESCRIPTION
**Pull request by sigurdm**
This seems to be the right thing to do if we want to emit
proto3 JSON.
Currently we were using the dart mangled name, that is also camel-case
but has a number of dart-specific disambiguations applied making it unsuitable for
interop.

If we want to parse JSON we need also know the field_name (with
underscores). That is for a later PR...

### CODE REVIEW

**Comment by sigurdm** [comment]
https://github.com/dart-lang/protobuf/issues/153

**Comment by sigurdm** [comment]
CC @nichite

File: protoc_plugin/pubspec.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 name: protoc_plugin
-version: 17.0.2
+version: 16.0.6
```
Comment (szakarias) [inline protoc_plugin/pubspec.yaml]: wrong version?

File: protoc_plugin/test/goldens/imports.pb [hunk 2]
```diff
@@ -13,9 +13,9 @@ import 'package2.pb.dart' as $2;
 
 class M extends $pb.GeneratedMessage {
   static final $pb.BuilderInfo _i = $pb.BuilderInfo('M')
-    ..a<M>(1, 'm', $pb.PbFieldType.OM, M.getDefault, M.create)
-    ..a<$1.M>(2, 'm1', $pb.PbFieldType.OM, $1.M.getDefault, $1.M.create)
-    ..a<$2.M>(3, 'm2', $pb.PbFieldType.OM, $2.M.getDefault, $2.M.create)
+    ..a<M>(1, '', $pb.PbFieldType.OM, M.getDefault, M.create)
```
Comment (szakarias) [inline protoc_plugin/test/goldens/imports.pb]: Consider adding a JSON name in the test that generates this file.

File: protoc_plugin/pubspec.yaml (hunk 1 above)
Comment (sigurdm) [reply protoc_plugin/pubspec.yaml]: Fixed

File: protoc_plugin/test/goldens/imports.pb (hunk 2 above)
Comment (sigurdm) [reply protoc_plugin/test/goldens/imports.pb]: Done

**Pull request merged**